### ISSUE TITLE
Add support for FetchContent and find_package

### ISSUE THREAD

**Issue by kroppt** [KhronosGroup/Vulkan-Utility-Libraries#92]
CC @gregjohnson2017

Disclaimer: I am not an expert on CMake usage.

TL;DR: Acquiring this library with [`FetchContent`](https://cmake.org/cmake/help/latest/module/FetchContent.html) prevents Vulkan SDK from building.

---

I took the example from [`tests/add_subdirectory`](https://github.com/KhronosGroup/Vulkan-Utility-Libraries/tree/87801a6c479399889eea37069f83437d0ea9458e/tests/add_subdirectory) and modified it to use [`FetchContent`](https://cmake.org/cmake/help/latest/module/FetchContent.html) instead of [`add_subdirectory`](https://cmake.org/cmake/help/latest/command/add_subdirectory.html). Running this works, but when you add a call to [`find_package`](https://cmake.org/cmake/help/latest/command/find_package.html), it fails:
```
CMake Error at CMakeLists.txt:34 (find_package):
  By not providing "FindVulkanUtilityLibraries.cmake" in CMAKE_MODULE_PATH
  this project has asked CMake to find a package configuration file provided
  by "VulkanUtilityLibraries", but CMake did not find one.

  Could not find a package configuration file provided by
  "VulkanUtilityLibraries" with any of the following names:

    VulkanUtilityLibrariesConfig.cmake
    vulkanutilitylibraries-config.cmake

  Add the installation prefix of "VulkanUtilityLibraries" to
  CMAKE_PREFIX_PATH or set "VulkanUtilityLibraries_DIR" to a directory
  containing one of the above files.  If "VulkanUtilityLibraries" provides a
  separate development package or SDK, be sure it has been installed.
```

This is important, because most projects, like the [Vulkan SDK](https://github.com/LunarG/VulkanTools), will [use `find_package` to locate this library](https://github.com/LunarG/VulkanTools/blob/faa20e76b9994554f67c45584b3a7e5006e1ed97/CMakeLists.txt#L69). So, projects that use [`FetchContent`](https://cmake.org/cmake/help/latest/module/FetchContent.html) for their Vulkan dependencies do not work.

I don't know how to solve this yet. I'm looking into [this answer](https://stackoverflow.com/a/65046283/5141875) to figure out what may need to be done. Any help would be appreciated.

---

Command used to run sample: `cmake -B build -G Ninja && cmake --build build`
Modified `CMakeLists.txt`:
```cmake
# Copyright 2023 The Khronos Group Inc. Copyright 2023 Valve Corporation
# Copyright 2023 LunarG, Inc.
#
# SPDX-License-Identifier: Apache-2.0
#
# Modified by Tyler Kropp
cmake_minimum_required(VERSION 3.17)

project(TEST_ADD_SUBDIRECTORY LANGUAGES C)

add_library(add_subdirectory_example STATIC)

# Test c99 support
target_compile_features(add_subdirectory_example PRIVATE c_std_99)

target_sources(add_subdirectory_example
               PRIVATE client.c vk_enum_string_helper.c vul_dispatch_table.c)

include(FetchContent)

FetchContent_Declare(
  vulkan-headers
  GIT_REPOSITORY "https://github.com/KhronosGroup/Vulkan-Headers.git"
  GIT_TAG "2634c96" # latest as of writing
)

FetchContent_Declare(
  vulkan-utility
  GIT_REPOSITORY "https://github.com/KhronosGroup/Vulkan-Utility-Libraries.git"
  GIT_TAG "87801a6" # latest as of writing
)

FetchContent_MakeAvailable(vulkan-headers vulkan-utility)

# Vulkan SDK uses find_package to look for this library
find_package(VulkanUtilityLibraries REQUIRED) # <<< this call fails

if(NOT TARGET Vulkan::LayerSettings)
  message(FATAL_ERROR "Vulkan::LayerSettings target not defined!")
endif()

if(NOT TARGET Vulkan::UtilityHeaders)
  message(FATAL_ERROR "Vulkan::UtilityHeaders target not defined!")
endif()

target_link_libraries(add_subdirectory_example PRIVATE Vulkan::LayerSettings
                                                       Vulkan::UtilityHeaders)
```

**Comment by juan-lunarg** [comment]
We do support `find_package`. It's actively used in VulkanProfiles and the ValidationLayers at this very moment. We even have tests for it in this repo.

FetchContent does NOT install packages. FetchContent by default clones a repo and calls add_subdirectory on it. IE making it a part of your project.

find_package relies on packages being built/installed by either a language package manager (vcpkg / conan for example) or a system package manager.

IE your example isn't correct.

```
include(FetchContent)

FetchContent_Declare(
  vulkan-headers
  GIT_REPOSITORY "https://github.com/KhronosGroup/Vulkan-Headers.git"
  GIT_TAG "2634c96" # latest as of writing
)

FetchContent_Declare(
  vulkan-utility
  GIT_REPOSITORY "https://github.com/KhronosGroup/Vulkan-Utility-Libraries.git"
  GIT_TAG "87801a6" # latest as of writing
)

FetchContent_MakeAvailable(vulkan-headers vulkan-utility)

if(NOT TARGET Vulkan::LayerSettings)
  message(FATAL_ERROR "Vulkan::LayerSettings target not defined!")
endif()

if(NOT TARGET Vulkan::UtilityHeaders)
  message(FATAL_ERROR "Vulkan::UtilityHeaders target not defined!")
endif()

target_link_libraries(add_subdirectory_example PRIVATE Vulkan::LayerSettings
                                                       Vulkan::UtilityHeaders)

# find_package(VulkanUtilityLibraries REQUIRED) # <<< this call fails because FetchContent doesn't install libraries. You already have the libraries when you called FetchContent_MakeAvailable
```

**Comment by kroppt** [comment]
> find_package relies on packages being built/installed by either a language package manager (vcpkg / conan for example) or a system package manager.

Thanks for the insight.

Looking more into that, it appears there is something new in CMake 3.24 that [integrates `FetchContent` with `find_package`](https://cmake.org/cmake/help/latest/guide/using-dependencies/index.html#fetchcontent-and-find-package-integration). By adding `OVERRIDE_FIND_PACKAGE` to my `FetchContent` call and making the name the same as the package, it all magically works!

Fixed sample:
```cmake
cmake_minimum_required(VERSION 3.24) # <<< update minimum version
...
FetchContent_Declare(
  vulkan-headers
  GIT_REPOSITORY "https://github.com/KhronosGroup/Vulkan-Headers.git"
  GIT_TAG "2634c96" # latest as of writing
)

FetchContent_Declare(
  VulkanUtilityLibraries # <<< name same as below
  GIT_REPOSITORY "https://github.com/KhronosGroup/Vulkan-Utility-Libraries.git"
  GIT_TAG "87801a6" # latest as of writing
  OVERRIDE_FIND_PACKAGE) # <<< add this

FetchContent_MakeAvailable(vulkan-headers VulkanUtilityLibraries) # <<< update name

# Vulkan SDK uses find_package to look for this library, and this call now succeeds
find_package(VulkanUtilityLibraries REQUIRED) # <<< this name here
...
```
